app/routes.py

```diff
@@ -1,2 +1,268 @@
+
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        new_task_response = { 
+            "task": new_task.get_task_dict()
+        }
+
+        return jsonify(new_task_response), 201
+
+    # GET REQUESTS
+    elif request.method == "GET":
+        title_query = request.args.get("title")
+        description_query = request.args.get("description")
+        if title_query:
+            tasks = Task.query.filter(Task.title.contains(title_query))
+        elif description_query:
+            tasks = Task.query.filter(Task.description.contains(description_query))
+        else:
+            tasks = Task.query.all()
+
+        # QUERY PARAMS (Ascending and Descending Order by Title)
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()) 
+        else:
+            tasks = Task.query.all()
+
+        task_response = []
+        for task in tasks:
+            task_response.append(task.get_task_dict())
+            
+        if task_response == []:
+            return jsonify(task_response), 200
+
+        return jsonify(task_response), 200
+
+# GET, PUT, DELETE ONE AT A TIME
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_one_task_at_a_time(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return jsonify(None), 404
+    
+    if request.method == "GET":
+        return {
+            "task": task.get_task_dict()
+        }, 20
+
+    elif request.method == "PUT":
+    
+        request_body = request.get_json()
+
+        task.title = request_body["title"]
+        task.description = request_body["description"]
+
+        db.session.commit()
+
+        updated_task_response = {
+            "task": task.get_task_dict()
+            }
+
+        return jsonify(updated_task_response), 200
+
+    elif request.method == "DELETE":
+    
+        db.session.delete(task)
+        db.session.commit()
+
+        delete_response = {"details": (f'Task {task_id} "{task.title}" successfully deleted')}
+
+        return jsonify(delete_response), 200
+
+# PATCH REQUESTS as either complete or not complete
+# endpoints are expressed w/out carrots for search specificity
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_tasks_complete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return jsonify(None), 404
+
+    # setting the completed_at with a datetime
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+
+    # Send Slack Message Using Slack API
+    slack_url = "https://slack.com/api/chat.postMessage"
+    bearer_token = os.environ.get("BEARER_TOKEN")
+    slack_channel = 'slack-api-test-channel'
+    text = f'Someone just completed the task {task.title}'
+    slack_user_name = "Starseed's Bot"
```

Comment:
you could also move your slack info into a helper function and call the function here to clean up this function